Project-II-Calculator/calculator/src/components/ButtonComponents/ActionButton.js

```diff
@@ -1,2 +1,16 @@
 import React from 'react';
-import './Button.css';
+
+function ActionButton(props) {
+    return (
+        <div className="action-btns">
+            {props.buttons.map(item => {
+                // equal sign 
+                if (item.text === '=') return <button key={item.text} className={`key-eq-sign ${item.buttonStyle}`}>{item.text}</button>;
+                
+                // math operators
+                else return <button key={item.text} className={`key-oper ${item.buttonStyle}`}>{item.text}</button>;
```

Comment:
I can see you got into the requirement for a mapped-over components to have unique keys, and the easiest way to accomplish that is how you do it here since the calculator buttons will all have unique text. in future projects where you might be working over an array of todo-list tasks though, for example, you might have duplicates of text. the two ways to still apply a unique key if each object in the data array has an `id` property starting at 0 and incrementing (which is how most databases will store them) or utilizing the optional `index` param in your map method by passing in `(item, index)` and thereby using the array's inherent index values to accomplish the same thing